source/grid/manifold.cc

```diff
@@ -87,6 +87,68 @@ get_new_point (const Quadrature<spacedim> &quad) const
   return p;
 }
 
+template <int dim, int spacedim>
+Point<spacedim>
+Manifold<dim, spacedim>::
+get_new_point (const std::vector<Point<spacedim> > &surrounding_points,
+               const std::vector<double>           &weights) const
+{
+  const double tol = 1e-10;
+  const unsigned int n_points = surrounding_points.size();
+
+  Assert(n_points > 0,
+         ExcMessage("There should be at least one point."));
+
+  Assert(n_points == weights.size(),
+         ExcMessage("There should be as many surrounding points as weights given."));
+
+  Assert(std::abs(std::accumulate(weights.begin(), weights.end(), 0.0)-1.0) < tol,
+         ExcMessage("The weights for the individual points should sum to 1!"));
+
+  // The number of points is small (at most 26 in 3D), therefore try to minize
+  // overhead and use a very simple search of O(N^2) to find the order in which
+  // to loop over points.
+  std::vector<unsigned int> permutation(n_points);
+  std::vector<bool> found(n_points,false);
```

Comment:
It can be any number of points if you just use a high enough polynomial degree for the mapping. 26 just happens the maximum you see in your application.